lib/matplotlib/widgets.py

```diff
@@ -295,12 +295,17 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
         dragging : bool, default: True
             If True the slider can be dragged by the mouse.
 
-        valstep : float, default: None
-            If given, the slider will snap to multiples of *valstep*.
+        valstep : float or array-like, default: None
+            If a float, the slider will snap to multiples of *valstep*.
+            If an array the slider will snap to the values in the array.
 
         orientation : {'horizontal', 'vertical'}, default: 'horizontal'
             The orientation of the slider.
 
+        initcolor : color or None, default: 'r'
+            The color of the line at the *valinit* position. If None then
+            no line will be drawn.
```

Comment:
Usually we use `None` to mean default. We also already have `'none'` to mean no colour; do we really need to support `None` here?

Reply:
I find using `None` to mean "don't draw" and colors otherwise to be the most intuitive api. I'd also worry about the discoverability of using `'none'` for example: I'm an advanced user but did not know that you could specify `'none'` as a color. 

> Usually we use None to mean default

I made the default `'r'`  as that's what was hardcoded, but I think it would make more sense for `None` to be the default and specifiying a color otherwise. I don't really see the usecase for drawing a line at the starting position, but I guess this edges into personal preference rather than logic and additionally is maybe backwards incompatible. What about switching the default to `None`?

Reply:
`None` means "default" unless we really can't help it.  I you want the default to be `"none"` I guess that could work for you....

Reply:
Gotcha - thanks.  It probably is the least friction to as `'r'` and not accept `None` so I'll do that + leave a note in the docstring to use `'none'` if you don't want it to be visible.